src/components/Editor/Runners/HtmlRunner/HtmlRunner.jsx

```diff
@@ -194,107 +202,105 @@ function HtmlRunner() {
           );
-          const matchingInternalHref = matchingRegexes(
-            allowedInternalLinks,
-            hrefNode.attrs.href,
-          );
-          if (
-            !matchingInternalHref &&
-            !matchingExternalHref &&
-            !parentTag(hrefNode, "head")
-          ) {
-            // eslint-disable-next-line no-script-url
-            hrefNode.setAttribute("href", "javascript:void(0)");
-            onClick =
-              "window.parent.postMessage({msg: 'ERROR: External link'})";
-          } else if (matchingExternalHref) {
-            onClick = `window.parent.postMessage({msg: 'Allowed external link', payload: { linkTo: '${hrefNode.attrs.href}' }})`;
-          }
+          hrefNode.setAttribute("href", projectFileBlob);
+        } else {
+          // eslint-disable-next-line no-script-url
+          hrefNode.setAttribute("href", "javascript:void(0)");
+          onClick = `window.parent.postMessage({msg: 'RELOAD', payload: { linkTo: '${projectFile.name}' }})`;
         }
-
-        if (onClick) {
-          hrefNode.removeAttribute("target");
-          hrefNode.setAttribute("onclick", onClick);
+      } else {
+        const matchingExternalHref = matchingRegexes(
+          allowedExternalLinks,
+          hrefNode.attrs.href,
+        );
+        const matchingInternalHref = matchingRegexes(
+          allowedInternalLinks,
+          hrefNode.attrs.href,
+        );
+        if (
+          !matchingInternalHref &&
+          !matchingExternalHref &&
+          !parentTag(hrefNode, "head")
+        ) {
+          // eslint-disable-next-line no-script-url
+          hrefNode.setAttribute("href", "javascript:void(0)");
+          onClick = "window.parent.postMessage({msg: 'ERROR: External link'})";
+        } else if (matchingExternalHref) {
+          onClick = `window.parent.postMessage({msg: 'Allowed external link', payload: { linkTo: '${hrefNode.attrs.href}' }})`;
         }
-      });
+      }
 
-      const replaceAttrsWithBlob = (attr = "src") => {
-        const srcNodes = indexPage.querySelectorAll(`[${attr}]`);
-        srcNodes.forEach((srcNode) => {
-          const projectImage = projectImages.filter(
-            (component) => component.filename === srcNode.attrs[attr],
-          );
-          const projectFile = projectCode.filter(
-            (file) => `${file.name}.${file.extension}` === srcNode.attrs[attr],
-          );
-          let src = "";
-          if (projectImage.length) {
-            src = projectImage[0].url;
-          } else if (projectFile.length) {
-            src = getBlobURL(
-              projectFile[0].content,
-              mimeTypes.lookup(
-                `${projectFile[0].name}.${projectFile[0].extension}`,
-              ),
-            );
-          }
-          srcNode.setAttribute(attr, src);
-        });
-      };
-      replaceAttrsWithBlob("src");
-      replaceAttrsWithBlob("data-src");
```

Comment:
🔧 this was the little functionality I added for the content team and it's needed. You could add the `attr` prop to your `replaceSrcNodes` function and do the same thing